processors/blockprocessor/block_processor.go

```diff
@@ -0,0 +1,386 @@
+			fmt.Errorf("prepareEvalResources() err: %w", err)
+	}
+
+	res := ledger.EvalForIndexerResources{
+		Accounts:  nil,
+		Resources: nil,
+		Creators:  make(map[ledger.Creatable]ledger.FoundAddress),
+	}
+
+	for index, foundAddress := range assetCreators {
+		creatable := ledger.Creatable{
+			Index: basics.CreatableIndex(index),
+			Type:  basics.AssetCreatable,
+		}
+		res.Creators[creatable] = foundAddress
+	}
+	for index, foundAddress := range appCreators {
+		creatable := ledger.Creatable{
+			Index: basics.CreatableIndex(index),
+			Type:  basics.AppCreatable,
+		}
+		res.Creators[creatable] = foundAddress
+	}
+
+	res.Accounts, res.Resources, err = prepareAccountsResources(l, block.Payset, assetCreators, appCreators)
+	if err != nil {
+		return ledger.EvalForIndexerResources{},
+			fmt.Errorf("prepareEvalResources() err: %w", err)
+	}
+
+	return res, nil
+}
+
+// Preload asset and app creators.
+func prepareCreators(l *indexerledger.LedgerForEvaluator, payset transactions.Payset) (map[basics.AssetIndex]ledger.FoundAddress, map[basics.AppIndex]ledger.FoundAddress, error) {
+	assetsReq, appsReq := accounting.MakePreloadCreatorsRequest(payset)
+
+	assets, err := l.GetAssetCreator(assetsReq)
+	if err != nil {
+		return nil, nil, fmt.Errorf("prepareCreators() err: %w", err)
+	}
+	apps, err := l.GetAppCreator(appsReq)
+	if err != nil {
+		return nil, nil, fmt.Errorf("prepareCreators() err: %w", err)
+	}
+
+	return assets, apps, nil
+}
+
+// Preload account data and account resources.
+func prepareAccountsResources(l *indexerledger.LedgerForEvaluator, payset transactions.Payset, assetCreators map[basics.AssetIndex]ledger.FoundAddress, appCreators map[basics.AppIndex]ledger.FoundAddress) (map[basics.Address]*ledgercore.AccountData, map[basics.Address]map[ledger.Creatable]ledgercore.AccountResource, error) {
+	addressesReq, resourcesReq :=
+		accounting.MakePreloadAccountsResourcesRequest(payset, assetCreators, appCreators)
+
+	accounts, err := l.LookupWithoutRewards(addressesReq)
+	if err != nil {
+		return nil, nil, fmt.Errorf("prepareAccountsResources() err: %w", err)
+	}
+	resources, err := l.LookupResources(resourcesReq)
+	if err != nil {
+		return nil, nil, fmt.Errorf("prepareAccountsResources() err: %w", err)
+	}
+
+	return accounts, resources, nil
+}
+
+// MakeLegacyProcessorHandlerFunction makes a legacy function that emulates original behavior of block processor
+func MakeLegacyProcessorHandlerFunction(proc *BlockProcessor, handler func(block *ledgercore.ValidatedBlock) error) func(cert *rpcs.EncodedBlockCert) error {
+	return func(cert *rpcs.EncodedBlockCert) error {
+		blockData, err := (*proc).Process(data.MakeBlockDataFromEncodedBlockCertificate(data.BlockData{}, cert))
+		if err != nil {
+			return err
+		}
+
+		proto, ok := config.Consensus[cert.Block.BlockHeader.CurrentProtocol]
+		if !ok {
+			return fmt.Errorf(
+				"cannot find proto version %s", cert.Block.BlockHeader.CurrentProtocol)
+		}
```

Comment:
We call the handler func at the end of this function, which I think is just the [importer.ImportBlock](https://github.com/algorand/indexer/blob/conduit/importer/importer.go#L24) function that also checks the protocol version. 

It's also called in `extractValidatedBlockAndPayset` during `Process`.

So I don't think this is necessary here.

Reply:
The problem is the conditional relies on either using the original certified block or the newly evaluated modified transactions.  The Process function returns the newly evaluated transactions without access to the original so I would have to use a conditional anyway.

I could always expose `extractValidatedBlockAndPayset` but I feel that would take away from the purpose of "only using public interfaces".  However this is a legacy adapter so maybe that is warranted.  Ping me on slack and we can call and discuss.